graylog2-web-interface/CONTRIBUTING.md

```diff
@@ -0,0 +1,194 @@
+# Contributing
+
+Please follow [the instructions on graylog.org](https://www.graylog.org/get-involved/).
+
+## Code of Conduct
+In the interest of fostering an open and welcoming environment, we as
+contributors and maintainers pledge to making participation in our project and
+our community a harassment-free experience for everyone, regardless of age, body
+size, disability, ethnicity, gender identity and expression, level of experience,
+nationality, personal appearance, race, religion, or sexual identity and
+orientation.
+
+Please read and understand the [Code of Conduct](https://github.com/Graylog2/graylog2-server/blob/master/CODE_OF_CONDUCT.md).
+
+## Conventions
+
+### Consistent Code Style
+- We use ESLint to detect some issues in our code. We mostly follow the [Airbnb Javascript style guide](https://github.com/airbnb/javascript) to write frontend code, with a few exceptions.
+- We maintain those custom rules in a package, which is part of [our graylog2-server repository](https://raw.githubusercontent.com/Graylog2/graylog2-server/master/graylog2-web-interface/packages/eslint-config-graylog/index.js).
+- Ensure you are seeing linter hints in you IDE and consider enabling the setting "fix linter hints on save" ([IntelliJ Docs](https://www.jetbrains.com/help/idea/eslint.html#ws_eslint_configure_run_eslint_on_save)).
+- There is a CI job, which checks if there are linter hints in changes files.
+- If you want to see all linter hints for changed (and committed) files locally, you can run `yarn lint:changes`. You will need to commit your changes first.
+- If you want to see linter hints for a specific file, you can run `lint:path /path/to/file`.
+
+### Naming
+#### Function Naming
+- Generally use a verb as a function name.
+
+#### Class Naming
+- Generally use a noun as a class name
+
+### Class vs functional components
+Small components should be functional components. When a component is more complex, you can decide which type of component
+you want to use (class or a functional component with react hooks). When you don’t have a preference, use a functional component.
+
+### Type definitions
+- We use TypeScript for new components, and we also define `PropType` definitions. Static types add better type support for the development, integrating with your IDE, while `PropType` definitions are present at runtime.
+- Names of unused parameters
+  - Sometimes you need to define a parameter which is not being used, for example when you just want to use the second parameter. To satisfy TypeScript the parameter name needs to be prefixed with underscore.
+  - Instead of defining just `_`, use a more meaningful name, like `_theParameterName`. (Related to this discussion){.
+- types.d.ts can have hidden errors, like missing imports. Rename it temporary to `types.d.ts` to detect the most obvious ones.
```

Comment:
renaming `types.d.ts` to `types.d.ts`?? I don't get it.

Reply:
Thanks, it should be "rename `types.d.ts` to `types.ts`".